### ISSUE TITLE
VarArray for arrays of custom Values

### ISSUE THREAD

**Issue by chrboe** [spf13/pflag#260]
I have a custom type implementing the `Value` interface, and I would like to pass an array of those values to my application.

There is `StringArray`, which allows passing in arrays of strings, and there is `Var`, which allows passing in custom `Value`s, but I was unable to find a "mix" of the two – something like `VarArray`.

Is there a reason why this is not yet implemented?

**Comment by tomasaschan** [comment]
This issue has not been updated in a long time. Is it still relevant?

(If I receive no answer before the end of the summer, I'll close it. Feel free to re-open if you come here after that happens and it is relevant to you - but if you do, please also comment and specify what outcome you're interested in!)

**Comment by chrboe** [comment]
Honestly, I had completely forgotten about this 🙂 

Seems like this is what I wanted that feature for back in the day: https://github.com/LINBIT/virter/commit/51993f87de574cfecc7c24d6a5be9ed49231fa9f#diff-ffc50958328bd76709f5607348540cbdead2088a1af28fbd3aa110e4abda82b2R196

I still agree with my past self that it would be nicer if pflag supported this directly, but the "workaround" has been working for us for over five years now, so it can't be that bad.

If you have a "maybe eventually"-type backlog, I guess this issue would belong there. If you prefer to close it, I don't mind either (but the 2 👍 on the original issue might)

**Comment by tomasaschan** [comment]
It seems this is pretty close in spirit to #359 - and, IIUC, that solving that would solve this too. I'm going to close this for now, but feel free to re-open if the discussion over in #359 results in an outcome that doesn't solve this problem.

**Comment by chrboe** [comment]
I agree, that seems like a good path to take. Nice to see some active maintenance going on in this project again, thanks for stepping up 💪
